src/backend/optimizer/plan/createplan.c

```diff
@@ -6688,10 +6688,13 @@ adjust_modifytable_flow(PlannerInfo *root, ModifyTable *node, List *is_split_upd
 				 */
 				if (optimizer_replicated_table_insert &&
```

Comment:
What happens if a user (or gpdb during table [redistribution](https://github.com/arenadata/gpdb/blob/5ee12881c09bdbd9375b27e50265442102796f14/src/backend/commands/tablecmds.c#L15716-L15722)) set `optimizer_replicated_table_insert` to false? I thought, this should be handled inside.

Reply:
> What happens if a user (or gpdb during table [redistribution](https://github.com/arenadata/gpdb/blob/5ee12881c09bdbd9375b27e50265442102796f14/src/backend/commands/tablecmds.c#L15716-L15722)) set `optimizer_replicated_table_insert` to false? I thought, this should be handled inside.

```sql
gpdb=# explain (costs off, verbose) insert into t_replicate_dst (id, i) select i, random() from t_replicate_src;
                          QUERY PLAN                          
--------------------------------------------------------------
 Insert on public.t_replicate_dst
   ->  Broadcast Motion 1:3  (slice1; segments: 1)
         Output: "*SELECT*".i, (("*SELECT*".random)::integer)
         ->  Subquery Scan on "*SELECT*"
               Output: "*SELECT*".i, "*SELECT*".random
               ->  Seq Scan on public.t_replicate_src
                     Output: t_replicate_src.i, random()
 Optimizer: Postgres query optimizer
(8 rows)

gpdb=# set optimizer_replicated_table_insert = off;
SET
gpdb=# explain (costs off, verbose) insert into t_replicate_dst (id, i) select i, random() from t_replicate_src;
                          QUERY PLAN                          
--------------------------------------------------------------
 Insert on public.t_replicate_dst
   ->  Broadcast Motion 1:3  (slice1; segments: 1)
         Output: "*SELECT*".i, (("*SELECT*".random)::integer)
         ->  Subquery Scan on "*SELECT*"
               Output: "*SELECT*".i, "*SELECT*".random
               ->  Seq Scan on public.t_replicate_src
                     Output: t_replicate_src.i, random()
 Optimizer: Postgres query optimizer
(8 rows)
```

Reply:
It's not an answer to my question. When `optimizer_replicated_table_insert` is set to off, this branch of code is ignored. And if Subquery Scan had SegmentGeneral distributuin, gpdb wouldn't add broadcast above it. So Subquery Scan got SingleQE distribution earlier and this query doesn't check my case.

Reply:
> When `optimizer_replicated_table_insert` is set to off, this branch of code is ignored

fixed